sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/VectorIndexSpec.java

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.Constants;
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Vector Indexes spec for Azure CosmosDB service.
+ */
+public final class VectorIndexSpec {
+
+    private final JsonSerializable jsonSerializable;
+    private String type;
+
+    /**
+     * Constructor
+     *
+     * @param path the path.
+     */
+    public VectorIndexSpec(String path) {
+        this.jsonSerializable = new JsonSerializable();
+        this.setPath(path);
+    }
+
+    /**
+     * Constructor.
+     */
+    public VectorIndexSpec() {
+        this.jsonSerializable = new JsonSerializable();
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param objectNode the object node that represents the included path.
+     */
+    public VectorIndexSpec(ObjectNode objectNode) { this.jsonSerializable = new JsonSerializable(objectNode); }
```

Comment:
does this constructor need to be public? We do our best to remove jackson from everywhere we can, and I know, it'd probably be impossible in Cosmos, but if this constructor can be kept internal, that might make our life easier in the future.

Reply:
This is a no-go. Ctor with ObjectNode (any Jackson type) should be internal - or use Map<String, Object> instead - that way we can do the conversion internally. We try to avoid any Jackson API in public surface area - like @lmolkova  mentioned we will probably never stop using Jackson internally - but we should not leak this into public surface area (one API in ChangefeedProcessor slipped through the cracks - but for new API this is the general guidance)

Reply:
Done